### DISCUSSION TITLE
Math coding interest groups

### DISCUSSION THREAD

**Discussion by predrag3141** [aidevnn/FastGoat#64] [General]
I would like to join or form an interest group for people that publish algorithms on github. Do you have or wish you had such a community?

**Comment by aidevnn** [comment]
Apologies for the delay! Unfortunately, GitHub web app notifications slipped past me, and I'm just catching up on your message now. 

I'd be happy to join your group!

**Reply by predrag3141** [reply]
Since you don't know of a group for people working on math projects, I created one: https://www.linkedin.com/groups/13122600.

**Reply by aidevnn** [reply]
Joined !!!

**Reply by predrag3141** [reply]
I'm going to post about PSLQ. I'd love to see a post about where you are headed with your group theory project. If I'm reading it correctly, that is the one thing you are working on. Though you may find I'm not always the first to understand engineering concepts, so I may have missed something.

**Reply by aidevnn** [reply]
Inside FastGoat, I used PSLQ to find the minimal polynomial for the splitting field of a given polynomial and then computed its Galois Group with full correspondences of subfields and subgroups.

The purpose was to discover algorithms and fine-tune them through a purely computational mathematics approach, with academic references, but it wasn't focused on concrete engineering use cases to solve real-world problems.

**Reply by aidevnn** [reply]
I experimented PSLQ as in your example on [pslq_vs_lwe](https://github.com/predrag3141/pslq_vs_lwe) by transpiling it from Python to C#, and I discovered the limitations of PSLQ. 
I've been pushing a 'craft-code' right now, and thanks for introducing me Learning With Errors Encryption.

**Comment by predrag3141** [comment]
Starting around [here](https://github.com/aidevnn/FastGoat/blob/main/FastGoat/Examples/AlgebraicIntegerRelationPSLQ.cs#L26), you use PSLQ to look for an integer relation among β and the powers of 𝛼, but with the imaginary part of each multiplied by π. Do I understand correctly that 𝛼 and β are roots of a polynomial, and PSLQ checks whether 𝛼 and β are related so one can be excluded as a root of the minimal polynomial?

Also, I'm curious why the imaginary parts of the numbers are multiplied by π.

**Reply by aidevnn** [reply]
It is an old technique from the LLL algorithm i used because i dont know how to adapt LLL for complex numbers or a tuple of real numbers. Substituting π for the complex unit I is effective solely for algebraic integers, as π is transcendental and lies outside the algebraic closure of the rationals.

**Reply by aidevnn** [reply]
It reminds me to experiment with PSLQ and Complex numbers without this trick.

**Reply by predrag3141** [reply]
So, using π suppresses unwanted solutions from LLL arising from relations between the real and imaginary parts of the input?

You will definitely avoid this problem without scaling by π when using the form of PSLQ that takes complex input.

**Reply by aidevnn** [reply]
The current implementation of abstract fields does not support the Hermitian product; therefore, the implementations of LLL and PSLQ are specific to real numbers and incorrect for complex numbers. It is helpful but faces many limitations beyond finite groups study.

Extending the current design to create a minimalistic Computer Algebra System will be impossible.

**Reply by aidevnn** [reply]
Delving into topics such as Elliptic curves, the PSLQ algorithm, or Infinite Abelian groups, with various hard-coded tricks, can aid in conceptualizing advanced designs before starting them.
